src/poetry/installation/chef.py

```diff
@@ -41,15 +41,10 @@ def should_prepare(self, archive: Path) -> bool:
     def is_wheel(self, archive: Path) -> bool:
         return archive.suffix == ".whl"
 
-    def get_cached_archive_for_link(self, link: Link) -> Link:
-        # If the archive is already a wheel, there is no need to cache it.
-        if link.is_wheel:
-            return link
-
+    def get_cached_archive_for_link(self, link: Link) -> Link | None:
```

Comment:
Can we make `get_cached_archive_for_link` return a `Path` instead of a `Link` (same for `get_cached_archives_for_link`)? That should make things even clearer.

Reply:
pushed a second commit that does this.  It does indeed tidy things up, but it also makes the MR quite a bit more substantial